File: Api/Api/Models/MovieModel.cs
namespace Api.Models
{
    public class MovieModel
	{
		public string HorizontalImageUrl { get; set; }
		public string VerticalImageUrl { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime PremiereDate { get; set; }
        public DateTime ShowingTillDate { get; set; }
        public int DurationInMinutes { get; set; }
        public Dictionary<DateTime, List<SessionModel>> Sessions { get; set; }
    }
}

File: cinema/src/app/models/movie.model.ts
import { Session } from "./session.model";

export class Movie {
    horizontalImageUrl: string; 
    verticalImageUrl: string;
    title: string;
    description: string;
    premiereDate: Date;
    showingTillDate: Date;
    durationInMinutes: number;
    sessions: { [date: string]: Session[] };
  
    constructor(
      horizontalImageUrl: string,
      verticalImageUrl: string,
      title: string,
      description: string,
      premiereDate: Date,
      showingTillDate: Date,
      durationInMinutes: number,
      sessions: { [date: string]: Session[] }
    ) {
      this.horizontalImageUrl = horizontalImageUrl;
      this.verticalImageUrl = verticalImageUrl;
      this.title = title;
      this.description = description;
      this.premiereDate = premiereDate;
      this.showingTillDate = showingTillDate;
      this.durationInMinutes = durationInMinutes;
      this.sessions = sessions;
    }
}
  
  
  